grails-app/services/com/unifina/service/ProductService.groovy

```diff
@@ -141,8 +146,18 @@ class ProductService {
 		}
 
 		Product product = findById(id, currentUser, Permission.Operation.WRITE)
+		if (product.isFree()) {
```

Comment:
Needs a comment here. Something like "Revoke public read permissions and grant them back after update" (+ why we need to do it?)